internal/dataplane/parser/translate_ingress_test.go

```diff
@@ -1177,44 +1177,86 @@ func TestGetDefaultBackendService(t *testing.T) {
 		}
 	}
 
-	t.Run("no ingresses", func(t *testing.T) {
-		_, ok := getDefaultBackendService([]netv1.Ingress{}, false)
-		require.False(t, ok, "expected no default backend service when no ingress has one defined")
-	})
-
-	t.Run("one ingress with default backend", func(t *testing.T) {
-		ingresses := []netv1.Ingress{
-			someIngress(time.Now(), "foo-svc"),
-		}
-
-		svc, ok := getDefaultBackendService(ingresses, false)
-		require.True(t, ok, "expected default backend service when one ingress has one defined")
-
-		assert.Equal(t, "foo-namespace.foo-svc.80", *svc.Name)
-		assert.Equal(t, "foo-svc.foo-namespace.80.svc", *svc.Host)
-		assert.NotNil(t, svc.Parent)
-
-		require.Len(t, svc.Routes, 1)
-		require.Len(t, svc.Routes[0].Paths, 1)
-		assert.Equal(t, "/", *svc.Routes[0].Paths[0])
-	})
-
-	t.Run("multiple ingresses with default backend", func(t *testing.T) {
-		now := time.Now()
-		ingresses := []netv1.Ingress{
-			someIngress(now.Add(time.Second), "newer"),
-			someIngress(now, "older"),
-		}
-
-		svc, ok := getDefaultBackendService(ingresses, false)
-		require.True(t, ok, "expected default backend service when there's at least one ingress with one defined")
-
-		assert.Equal(t, "foo-namespace.older.80", *svc.Name, "expected older ingress to be selected")
-		assert.Equal(t, "older.foo-namespace.80.svc", *svc.Host)
-		assert.NotNil(t, svc.Parent)
+	now := time.Now()
+	testCases := []struct {
+		name             string
+		ingresses        []netv1.Ingress
+		expressionRoutes bool
+		// expected results.
+		haveBackendService bool
+		serviceName        string
+		serciceHost        string
```

Comment:
Might as well add a prefix of expected or enclose it in an `expected` field which would hold all of those.

Reply:
done